src/iron_math.py

```diff
@@ -1,16 +1,35 @@
 import math
+import arcade
 from typing import Tuple
 
 
-def rotate_vec2(point: Tuple[float, float], radians: float, /) -> float:
+def rotate_vec(point: Tuple[float, float], radians: float, /) -> float:
     x, y = point
     s = math.sin(radians)
     c = math.cos(radians)
     return (c * x - s * y, s * x + c * y)
 
 
-def add_vec2(point: Tuple[float, float], translation: Tuple[float, float], /) -> float:
+def add_vec(point: Tuple[float, float], translation: Tuple[float, float], /) -> float:
     return (point[0] + translation[0], point[1] + translation[1])
 
-def scale_vec2(point: Tuple[float, float], factor: float, /) -> float:
+
+def scale_vec(point: Tuple[float, float], factor: float, /) -> float:
     return (point[0] * factor, point[1] * factor)
+
+
+def offset_sprite_from(
+    child: arcade.Sprite, parent: arcade.Sprite, offset: Tuple[float, float, float]
+):
+    child.angle = parent.angle + offset[2]
+    child.position = add_vec(parent.position, rotate_vec(offset[:2], parent.radians))
+
+
+def combine_transform(parent: arcade.Sprite, offset: Tuple[float, float, float]):
+    angle = parent.angle + offset[2]
+    position = add_vec(parent.position, rotate_vec(offset[:2], parent.radians))
+    return (position[0], position[1], angle)
+
+
+def polar_to_vec(speed: float, radians: float):
```

Comment:
does this fcn only work for converting polar notation of speed to Cartesian coordinates?  The more familiar name for the first parameter would be `radius`, or perhaps `magnitude`, though that would be confusing if negative values are allowed

Reply:
I like `magnitude`.  Matches Unity's API: https://docs.unity3d.com/ScriptReference/Vector2-magnitude.html

No strong opinion from me on the negative number thing.  Negative radius doesn't make sense either.  So if someone does throw a negative number at this function, the outcome seems clear to me.  But we can add a docstring saying "negative numbers are equivalent to adding Pi to the radians, and flip the vector 180"

Reply:
I think polar is the right term, but vec is confusing because a vector could be a polar vector or cartesian vector. How about `polar_to_cartesian`?

Reply:
negative radius is important for things like angular momentum, where the axis of the radius is perpendicular to the plane of rotation, though at the moment, this fcn is only for converting positional vectors so if we want to restrain the first arg to be >= 0, I think it's reasonable.

Reply:
> I think polar is the right term, but vec is confusing because a vector could be a polar vector or cartesian vector. How about `polar_to_cartesian`?

I think that is a more descriptive name, but given that "vector" is a term of art meaning Cartesian position vector, I don't think it's a mandatory change.